### PR TITLE
Add REST client CodeLens for additional annotations

### LINKED ISSUE #94: REST client CodeLens only shows up for GET annotations.
Currently the MicroProfile REST Client CodeLens only show up for paths annotated with `@GET`.

I think it makes sense to display them for other annotations like POST or PUT even though they cannot be accessed from the browser. (If possible maybe we could make CodeLens for POST/PUT etc. requests not clickable?) 

IE. if u have a rest client with just a POST endpoint, you might still want the CodeLens to appear so you can quickly see the full URL that is configured by your mp-config.properties file.

### PR DESCRIPTION
**Pull request by JessicaJHee**
Fixes #94

Signed-off-by: Jessica He <jhe@redhat.com>

### CODE REVIEW

**Comment by rgrunber** [comment]
test this please.

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/jaxrs/java/JaxRsCodeLensParticipant.java [hunk 1]
```diff
@@ -133,7 +133,7 @@ private static void collectURLCodeLenses(IJavaElement[] elements, String rootPat
 				// @DELETE, @PUT
 				// JAX-RS
 				// annotation
-				if (isJaxRsRequestMethod(method) && Flags.isPublic(method.getFlags())) {
+				if (isClickableJaxRsRequestMethod(method) && Flags.isPublic(method.getFlags())) {
```
Comment (datho7561) [inline microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/jaxrs/java/JaxRsCodeLensParticipant.java:136]: What do you think about also providing the unclickable urls for the application's own REST endpoints in this PR? When I read the issue initially, that's what I thought it was talking about.

**Review by datho7561 — APPROVED** [review]
This works well, and the code looks good! I think we are going to hold off on merging this until after the release, although we should discuss that.

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/jaxrs/java/JaxRsCodeLensParticipant.java (hunk 1 above)
Comment (JessicaJHee) [reply microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/jaxrs/java/JaxRsCodeLensParticipant.java:136]: I think it makes sense to do that as well. @rgrunber what do you think?

**Comment by rgrunber** [comment]
Change works well for me. Do we care about `HEAD` in the event a user decides to customize ? or is that basically overkill.

jaxrs-head-support

**Comment by JessicaJHee** [comment]
> Change works well for me. Do we care about `HEAD` in the event a user decides to customize ? or is that basically overkill.

I didn't include it initially because I thought it wasn't very useful to show, but I can also add it to be consistent since it is allowed?

**Comment by rgrunber** [comment]
> > Change works well for me. Do we care about `HEAD` in the event a user decides to customize ? or is that basically overkill.
> 
> I didn't include it initially because I thought it wasn't very useful to show, but I can also add it to be consistent since it is allowed?

If it's easy enough, I would add it. It also isn't clickable, mainly because I don't think a web browser would render it.

**Review by rgrunber — APPROVED** [review]

**Pull request merged**